source/error.c

```diff
@@ -182,172 +181,24 @@ void aws_register_error_info(const struct aws_error_info_list *error_info) {
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_PRIORITY_QUEUE_EMPTY,
-        "Attempt to pop an item from an empty queue."),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_PRIORITY_QUEUE_BAD_NODE,
-        "Bad node handle passed to remove."),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_HASHTBL_ITEM_NOT_FOUND,
-        "Item not found in hash table."),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_INVALID_DATE_STR,
-        "Date string is invalid and cannot be parsed."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_INVALID_ARGUMENT,
-        "An invalid argument was passed to a function."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_RANDOM_GEN_FAILED,
-        "A call to the random number generator failed. Retry later."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_MALFORMED_INPUT_STRING,
-        "An input string was passed to a parser and the string was incorrectly formatted."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_UNIMPLEMENTED,
-        "A function was called, but is not implemented."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_INVALID_STATE,
-        "An invalid state was encountered."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_ENVIRONMENT_GET,
-        "System call failure when getting an environment variable."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_ENVIRONMENT_SET,
-        "System call failure when setting an environment variable."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_ENVIRONMENT_UNSET,
-        "System call failure when unsetting an environment variable."
-    ),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_SYS_CALL_FAILURE,
-        "System call failure"),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_FILE_INVALID_PATH,
-        "Invalid file path."),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_MAX_FDS_EXCEEDED,
-        "The maximum number of fds has been exceeded."),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_NO_PERMISSION,
-        "User does not have permission to perform the requested action."),
-    AWS_DEFINE_ERROR_INFO_COMMON(
-        AWS_ERROR_STREAM_UNSEEKABLE,
-        "Stream does not support seek operations"),
-};
-/* clang-format on */
-
-static struct aws_error_info_list s_list = {
-    .error_list = errors,
-    .count = AWS_ARRAY_SIZE(errors),
-};
-
-void aws_load_error_strings(void) {
-    if (!s_error_strings_loaded) {
-        s_error_strings_loaded = 1;
-        aws_register_error_info(&s_list);
+void aws_unregister_error_info(const struct aws_error_info_list *error_info) {
+    AWS_ASSERT(error_info);
+    AWS_ASSERT(error_info->error_list);
+    AWS_ASSERT(error_info->count);
+
+    const int min_range = error_info->error_list[0].error_code;
+    const int slot_index = min_range >> SLOT_DIV_SHIFT;
+
+    AWS_ASSERT(slot_index < AWS_MAX_ERROR_SLOTS && slot_index >= 0);
```

Comment:
If `AWS_FATAL_ASSERT()` could take an error message string, these could both be combined into a single `AWS_FATAL_ASSERT`

Reply:
We're logging right below if this condition is bad, so I'm just gonna nuke this.